magicctapipe/scripts/lst1_magic/semi_automatic_scripts/merging_runs.py

```diff
@@ -0,0 +1,571 @@
+                        ]
+                        for r in runs:
+                            if (len(glob.glob(f"{MAGIC_DL1_dir}/M1/{i}/{r}")) > 0) and (
+                                len(glob.glob(f"{MAGIC_DL1_dir}/M2/{i}/{r}")) > 0
+                            ):
+                                if not os.path.exists(
+                                    f"{MAGIC_DL1_dir}/Merged/{i}/Merged"
+                                ):
+                                    os.mkdir(f"{MAGIC_DL1_dir}/Merged/{i}/Merged")
+                                if not os.path.exists(
+                                    f"{MAGIC_DL1_dir}/Merged/{i}/Merged/logs"
+                                ):
+                                    os.mkdir(f"{MAGIC_DL1_dir}/Merged/{i}/Merged/logs")
+                                f.write(
+                                    f"conda run -n {env_name} merge_hdf_files --input-dir {MAGIC_DL1_dir}/Merged/{i}/{r} --output-dir {MAGIC_DL1_dir}/Merged/{i}/Merged --run-wise >{MAGIC_DL1_dir}/Merged/{i}/Merged/logs/merge_{i}_{r}_"
+                                    + "${SLURM_JOB_ID}.log \n"
+                                )
+            else:
+                dates = [
+                    os.path.basename(x) for x in glob.glob(f"{MAGIC_DL1_dir}/M1/*")
+                ]
+                for i in dates:
+                    if not os.path.exists(f"{MAGIC_DL1_dir}/Merged/{i}/Merged"):
+                        continue
+                    if not os.path.exists(f"{MAGIC_DL1_dir}/Merged/Merged_{i}"):
+                        os.mkdir(
+                            f"{MAGIC_DL1_dir}/Merged/Merged_{i}"
+                        )  # Creating a merged directory for each night
+                    if (
+                        len(glob.glob(f"{MAGIC_DL1_dir}/Merged/{i}/Merged/*M1*.h5")) > 0
+                    ) and (
+                        len(glob.glob(f"{MAGIC_DL1_dir}/Merged/{i}/Merged/*M2*.h5")) > 0
+                    ):
+                        if not os.path.exists(
+                            f"{MAGIC_DL1_dir}/Merged/Merged_{i}/logs"
+                        ):
+                            os.mkdir(f"{MAGIC_DL1_dir}/Merged/Merged_{i}/logs")
+                        f.write(
+                            f"conda run -n {env_name} merge_hdf_files --input-dir {MAGIC_DL1_dir}/Merged/{i}/Merged --output-dir {MAGIC_DL1_dir}/Merged/Merged_{i} >{MAGIC_DL1_dir}/Merged/Merged_{i}/logs/merge_night_{i}_"
+                            + "${SLURM_JOB_ID}.log \n"
+                        )
+
+
+def mergeMC(target_dir, identification, env_name, cwd, source_name):
+
+    """
+    This function creates the bash scripts to run merge_hdf_files.py in all MC runs.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    identification : str
+        Tells which batch to create. Options: protons, gammadiffuse
+    env_name : str
+        Name of the environment
+    cwd : Path
```

Comment:
I guess you do not have to pass this cwd as a parameter just getcwd before changing the file (unless there is some mess up of the directories when moving between the data and MCs)

Reply:
Maybe fixed